File: Kattis/dartscores.java
import java.util.*;

public class dartscores {

    static class circle {
        private int radius;

        public circle(int radius) {
            this.radius = radius;
        }

        public boolean isInside(int x, int y) {
            return (Math.pow(x, 2) + Math.pow(y, 2)) <= Math.pow(this.radius, 2); 
        }
    }

    public static List<circle> circles = new ArrayList<circle>();

    public static int solve(int x, int y) {
        int count = 1;

        for(circle c : circles) {
            if(c.isInside(x, y))
                break;
            count++;
        }

        return (count > 12) ? 0 : (11 - count);
    }

    public static void main(String[] args) {

        for(int i = 20; i <= 200; i += 20)
            circles.add(new circle(i));

        Scanner sc = new Scanner(System.in);
        sc.useDelimiter("\\n");
        int cases = sc.nextInt();

        while(cases-- > 0) {
            int darts = sc.nextInt();
            int ans = 0;

            for(int i = 0; i < darts; i++) {
                String[] parse = sc.next().split(" ");
                int x = Integer.parseInt(parse[0]);
                int y = Integer.parseInt(parse[1]);

                ans += solve(x, y);
            }

            System.out.println(ans);
        }
    }
}

File: Kattis/dartscores.c
#include <math.h>
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

int* circles;

bool isInside(int x, int y, int radius) {
	return (pow(x, 2) + pow(y, 2)) <= pow(radius, 2); 
}

int solve(int x, int y) {
	int count = 1, i = 0;

	for(; i < 10; i++) {
		int radius = *(circles + i);
		if(isInside(x, y, radius))
			break;
		count++;
	}

	return (count > 12) ? 0 : (11 - count);
}

int main(int argc, char const *argv[]) {
	circles = (int*) malloc(sizeof(int) * 10);

	int radius = 20, i = 0;
	for(; radius <= 200; radius += 20, i++)
		*(circles + i) = radius;

	int testcases;
	scanf("%d", &testcases);

	while(testcases--) {
		int darts, i = 0, ans = 0;
		scanf("%d", &darts);
		
		for(; i < darts; i++) {
			int x, y;
			scanf("%d %d", &x, &y);
			ans += solve(x, y);
		}

		printf("%d\n", ans);
	} 

	return 0;
}